infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ImagePullSecretProvisionerTest.java

```diff
@@ -0,0 +1,215 @@
+  @Mock
+  private NonNamespaceOperation namespaceOperation;
+
+  @Mock private Pod pod;
+  @Mock private PodSpec podSpec;
+  private LocalObjectReference existingImagePullSecretRef =
+      new LocalObjectReferenceBuilder().withName("existing").build();
+  private LocalObjectReference newImagePullSecretRef =
+      new LocalObjectReferenceBuilder().withName(ImagePullSecretProvisioner.SECRET_NAME).build();
+
+  private ImagePullSecretProvisioner imagePullSecretProvisioner;
+
+  @BeforeMethod
+  public void setup() throws InfrastructureException {
+    when(userSpecificDockerRegistryCredentialsProvider.getCredentials()).thenReturn(authConfigs);
+
+    when(k8sEnv.getPods()).thenReturn(ImmutableMap.of("wksp", pod));
+    when(pod.getSpec()).thenReturn(podSpec);
+    when(podSpec.getImagePullSecrets()).thenReturn(ImmutableList.of(existingImagePullSecretRef));
+
+    when(runtimeIdentity.getWorkspaceId()).thenReturn("workspaceId");
+
+    when(namespaceFactory.create(anyString())).thenReturn(namespace);
+    when(namespace.getName()).thenReturn("namespaceName");
+
+    when(clientFactory.create()).thenReturn(kubernetesClient);
+    when(clientFactory.create(anyString())).thenReturn(kubernetesClient);
+
+    doReturn(mixedOperation).when(kubernetesClient).secrets();
+    when(mixedOperation.inNamespace(anyString())).thenReturn(namespaceOperation);
+
+    imagePullSecretProvisioner =
+        new ImagePullSecretProvisioner(
+            userSpecificDockerRegistryCredentialsProvider, clientFactory, namespaceFactory);
+  }
+
+  @Test
+  public void dontDoAnythingIfNoPrivateRegistries() throws Exception {
+    when(authConfigs.getConfigs()).thenReturn(Collections.emptyMap());
+
+    imagePullSecretProvisioner.provision(k8sEnv, runtimeIdentity);
+
+    verifyZeroInteractions(namespaceOperation);
+    verifyZeroInteractions(podSpec);
+  }
+
+  private static class TestAuthConfig implements AuthConfig {
+    private String username;
+    private String password;
+
+    @Override
+    public String getUsername() {
+      return username;
+    }
+
+    @Override
+    public void setUsername(String username) {
+      this.username = username;
+    }
+
+    @Override
+    public AuthConfig withUsername(String username) {
+      setUsername(username);
+      return this;
+    }
+
+    @Override
+    public String getPassword() {
+      return password;
+    }
+
+    @Override
+    public void setPassword(String password) {
+      this.password = password;
+    }
+
+    @Override
+    public AuthConfig withPassword(String password) {
+      setPassword(password);
+      return this;
+    }
+  }
+
+  private String buildAuth(String username, String password) {
+    return Base64.getEncoder()
+        .encodeToString(
+            new StringBuilder()
+                .append(username)
+                .append(':')
+                .append(password)
+                .toString()
+                .getBytes());
+  }
+
+  @SuppressWarnings("unchecked")
```

Comment:
If you use suppressing of warnings, please, add a comment to the doc that explains why it is safe to suppress warnings.

Reply:
BTW it's not good practice to use SuppressWarnings for the whole method.